Operators/Types/lib/string/IndexOf.cs

```diff
@@ -24,13 +24,21 @@ public IndexOf()
         private void Update(EvaluationContext context)
         {
             string searchPattern = SearchPattern.GetValue(context);
-            if (string.IsNullOrEmpty(searchPattern))
+            string originalString = OriginalString.GetValue(context);
+            if (string.IsNullOrEmpty(searchPattern) || string.IsNullOrEmpty(originalString))
             {
                 Index.Value = -1;
                 return;
             }
-            string originalString = OriginalString.GetValue(context);
-            Index.Value = originalString.IndexOf(searchPattern);
+
+            try
+            {
+                Index.Value = originalString.IndexOf(searchPattern);
+            }
+            catch (Exception)
```

Comment:
Sorry for the confusion. In that case, just checking for null in the line above would have been completely sufficient :-)